tests/models/test_schema.py

```diff
@@ -0,0 +1,162 @@
+import glob
+import json
+from copy import deepcopy
+from pathlib import Path
+from typing import Any
+
+import pytest
+
+from mex.common.models import EXTRACTED_MODEL_CLASSES_BY_NAME
+from mex.common.transform import dromedary_to_kebab
+from mex.common.types.identifier import MEX_ID_PATTERN, Identifier
+
+# TODO: find a cleaner way to get to the mex-model JSON schemas
+SPECIFIED_SCHEMA_PATH = Path(".venv", "src", "mex-model", "schema", "entities")
+
+GENERATED_SCHEMAS = dict(
+    sorted(
+        {
+            name.removeprefix("Extracted"): model.model_json_schema()
+            for name, model in EXTRACTED_MODEL_CLASSES_BY_NAME.items()
+        }.items()
+    )
+)
+SPECIFIED_SCHEMAS = dict(
+    sorted(
+        {
+            schema["title"].replace(" ", ""): schema
+            for file_name in glob.glob(str(SPECIFIED_SCHEMA_PATH / "*.json"))
+            if (schema := json.load(open(file_name, encoding="utf-8")))
+            and not schema["title"].startswith("Concept")
+        }.items()
+    )
+)
+ENTITY_TYPES_AND_FIELD_NAMES_BY_FQN = {
+    f"{entity_type}.{field_name}": (entity_type, field_name)
+    for entity_type, schema in SPECIFIED_SCHEMAS.items()
+    for field_name in schema["properties"]
+}
+
+
+def test_entity_types_match_spec() -> None:
+    assert list(GENERATED_SCHEMAS) == list(SPECIFIED_SCHEMAS)
+
+
+@pytest.mark.parametrize(
+    ("generated", "specified"),
+    zip(GENERATED_SCHEMAS.values(), SPECIFIED_SCHEMAS.values()),
+    ids=GENERATED_SCHEMAS,
+)
+def test_field_names_match_spec(
+    generated: dict[str, Any], specified: dict[str, Any]
+) -> None:
+    assert set(generated["properties"]) == set(specified["properties"])
+
+
+@pytest.mark.parametrize(
+    ("generated", "specified"),
+    zip(GENERATED_SCHEMAS.values(), SPECIFIED_SCHEMAS.values()),
+    ids=GENERATED_SCHEMAS,
+)
+def test_required_fields_match_spec(
+    generated: dict[str, Any], specified: dict[str, Any]
+) -> None:
+    assert set(generated["required"]) == set(specified["required"])
+
+
+@pytest.mark.parametrize(
+    ("entity_type", "field_name"),
+    ENTITY_TYPES_AND_FIELD_NAMES_BY_FQN.values(),
+    ids=ENTITY_TYPES_AND_FIELD_NAMES_BY_FQN.keys(),
+)
+def test_field_defs_match_spec(entity_type: str, field_name: str) -> None:  # noqa: C901
```

Comment:
I tested it and I can see what it's doing, but unfortunately I did not really crack the function as whole